### PR TITLE
[AMORO-2033]: The official docker image

### LINKED ISSUE #868: [Improvement]: Docker Images for quick-start are too large
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/arctic/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?
Docker Images for quick-start are too large, this is a burden for users to use quick-start.
截屏2022-12-07 10 10 48
datanode and namenode each occupy 1.47G.
截屏2022-12-07 10 14 18
ams 1.4G
image
flink 1.6g

### How should we improve?

Make the parent image from open-jdk to slim version.
Remove unnecessary tools in images.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Subtasks

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### LINKED ISSUE #2033: [Feature]: Official Amoro Docker Image.
### Description

According to the discussion in https://github.com/NetEase/amoro/discussions/2022 , the following plan is adopted to provide the mirror:

Production environment mirror:

- `arctic163/amoro`
-  `arcitc163/optimizer-flink1.1x`

Quick-start mirror:

- `arctic163/amoro-demo`

Based on this, Helm installation script and deployment documentation will be provided.

### Use case/motivation

_No response_

### Describe the solution

_No response_

### Subtasks

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by baiyangtx**
## Why are the changes needed?
 
Close #2033.  Close #868

## Brief change log

- Change `docker/ams` to `docker/quickdemo`.
- Add `docker/amoro` as official image for production environment deployment.
- Add build script for those images.

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [X] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (no)

### CODE REVIEW

**Comment by baiyangtx** [comment]
Quick Demo should also changed:  https://github.com/NetEase/amoro-site/pull/1

File: docker/build.sh [hunk 1]
```diff
@@ -18,52 +18,53 @@
...
+    optimizer-flink         Build official Amoro optimizer deployed with flink engine for production environments.
+    amoro                   Build official Amoro image used for production environments.
 
 Options:
     --flink-version         Flink binary release version, default is 1.15.3, format must be x.y.z
     --hadoop-version        Hadoop binary release version, default is 2.10.2, format must be x.y.z
     --apache-archive        Apache Archive url, default is https://archive.apache.org/dist
     --debian-mirror         Mirror url of debian, default is http://deb.debian.org
     --optimizer-job         Location of optimizer job
+    --tag                   Tag for amoro/optimizer-flink/quickstart image.
```
Comment (HuangFru) [inline docker/build.sh:54]: quickdemo image

File: dist/src/main/arctic-bin/bin/ams.sh [hunk 2]
```diff
@@ -142,6 +147,9 @@ case "$1" in
             start
         fi
         ;;
+    start-foreground)
```
[the remark this answers is not in the document]
Comment (baiyangtx) [reply dist/src/main/arctic-bin/bin/ams.sh:150]: arctic163/amoro will use this to start AMS.  so AMS will be the main process of docker container.

File: docker/demo-cluster.sh [hunk 3]
```diff
@@ -18,20 +18,15 @@
 #
 
 
-PROJECT_VERSION=latest
+AMORO_TAG=master-snapshot
```
Comment (HuangFru) [inline docker/demo-cluster.sh:21]: Does this mean we will usually update the image with the tag "master-snapshot"?

File: dist/src/main/arctic-bin/conf/log4j2.xml [hunk 4]
```diff
@@ -112,6 +118,8 @@
 <!--            <AppenderRef ref="debugFileAppender" level="debug"/>-->
             <AppenderRef ref="infoFileAppender" level="info"/>
             <AppenderRef ref="errorFileAppender" level="error"/>
+            <!-- SET CONSOLE_LOG_LEVEL=info to print console logger in docker -->
+            <AppenderRef ref="consoleAppender" level="${CONSOLE_LEVEL}" />
```
[the remark this answers is not in the document]
Comment (baiyangtx) [reply dist/src/main/arctic-bin/conf/log4j2.xml:122]: docker logs will inspect the console log. and the file logs is needed to rolling up.

File: docker/demo-cluster.sh (hunk 3 above)
Comment (baiyangtx) [reply docker/demo-cluster.sh:21]: Yes,  I will configure a github ci, once PR merge to master, a new image will be build and push.

File: docker/amoro/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+
+# Usage:
+#   Run the docker command below under project dir.
+#      docker build \
+#        --file docker/amoro/Dockerfile \
+#        --tag arctic163/amoro:tagname
+#        .
+
+FROM openjdk:8u332 as builder
```
[the remark this answers is not in the document]
Comment (baiyangtx) [reply docker/amoro/Dockerfile:26]: yes.

[the remark this answers is not in the document]
Comment (baiyangtx) [reply docker/amoro/Dockerfile:26]: I used a builder container here to reduce the final image size. I need to unzip the zip file in the builder container in advance and remove flink-optimizer.jar, which will significantly reduce the image size.

File: docker/amoro/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+ADD . /workspace/amoro
+WORKDIR /workspace/amoro
+
+RUN AMORO_VERSION=`cat pom.xml | grep 'amoro-parent' -C 3 | grep -Eo '<version>.*</version>' | awk -F'[><]' '{print $3}'` \
```
[the remark this answers is not in the document]
Comment (HuangFru) [reply docker/amoro/Dockerfile:31]: Does this need a doc to explain?

File: docker/amoro/Dockerfile (hunk 5 above)
[the remark this answers is not in the document]
Comment (HuangFru) [reply docker/amoro/Dockerfile:26]: So #868 could also be closed by this PR.

[the remark this answers is not in the document]
Comment (baiyangtx) [reply docker/amoro/Dockerfile:26]: Yes.

**Review by HuangFru — APPROVED** [review]
LGTM.

File: docker/amoro/Dockerfile (hunk 6 above)
[the remark this answers is not in the document]
Comment (baiyangtx) [reply docker/amoro/Dockerfile:34]: The build.sh has automatically handled the path issue.

[the remark this answers is not in the document]
Comment (baiyangtx) [reply docker/amoro/Dockerfile:34]: has automatically handled the path issue.

File: docker/build.sh (hunk 1 above)
Comment (baiyangtx) [reply docker/build.sh]: fixed

**Comment by czy006** [comment]
[openjdk docker hub](https://hub.docker.com/_/openjdk) image is officially deprecated and all users are recommended to find and use suitable replacements ASAP.What about use [eclipse-temurin](https://hub.docker.com/_/eclipse-temurin) to replace it?

cc @zhoujinsong @baiyangtx

**Review by czy006 — APPROVED** [review]

**Comment by HuangFru** [comment]
> [openjdk docker hub](https://hub.docker.com/_/openjdk) image is officially deprecated and all users are recommended to find and use suitable replacements ASAP.What about use [eclipse-temurin](https://hub.docker.com/_/eclipse-temurin) to replace it?
> 
> cc @zhoujinsong @baiyangtx

Good idea, I find that Kyuubi and Trino also use eclipse-temurin to build the image.

**Comment by baiyangtx** [comment]
> [openjdk docker hub](https://hub.docker.com/_/openjdk) image is officially deprecated and all users are recommended to find and use suitable replacements ASAP.What about use [eclipse-temurin](https://hub.docker.com/_/eclipse-temurin) to replace it?
> 
> cc @zhoujinsong @baiyangtx

Great suggestion, I will adjust the base image accordingly.

**Comment by baiyangtx** [comment]
The base image changed to eclipse-temurin:8-jdk-jammy

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Pull request merged**